src/Microsoft.ML.Core/Utilities/NormStr.cs

```diff
@@ -18,37 +19,28 @@ namespace Microsoft.ML.Runtime.Internal.Utilities
     /// </summary>
     public sealed class NormStr
     {
-        public readonly string Value;
+        public readonly ReadOnlyMemory<char> Value;
         public readonly int Id;
         private readonly uint _hash;
 
         /// <summary>
         /// NormStr's can only be created by the Pool.
         /// </summary>
-        private NormStr(string str, int id, uint hash)
+        private NormStr(ReadOnlyMemory<char> str, int id, uint hash)
         {
-            Contracts.AssertValue(str);
-            Contracts.Assert(id >= 0 || id == -1 && str == "");
+            Contracts.Assert(id >= 0 || id == -1 && str.IsEmpty);
             Value = str;
             Id = id;
             _hash = hash;
         }
 
-        public override string ToString()
-        {
-            return Value;
-        }
+        public override string ToString() => Value.ToString();
```

Comment:
I'm a bit concerned about this change.  Every time someone calls the ToString() method of NormStr it will copy from the ROM<char> to a new string.

I see this used to just hold on to `string` and not `DvText`.  Should we change it back to hold on to `string` instead?  (and do the copy on insertion into the pool, as opposed to potentially every time it is accessed through `NormStr.ToString()`)

/cc @TomFinley